### ISSUE TITLE
Drop Down Title does not change when selecting from multiple Request Body Examples

### ISSUE THREAD

**Issue by jpjpjp** [scalar/scalar#4186]
### What happens?

Scalar is beginning to use the examples from the spec to provide dropdowns with pre-populated request bodies in the documentation that is being generated with the express-api-reference package.   However, I've noticed that the name of the example being selected always shows the default example name, even if the user selects one of the other examples.   A video is worth 1000 words...

https://github.com/user-attachments/assets/c06b7b16-4cf8-4b40-8ebe-b997134853fa

### What did you expect to happen?

I would expect the title in the drop down box to reflect the example that was selected.

It would be awesome if this gets addressed if someone could respond to my Discord message about this: https://discord.com/channels/1135330207960678410/1135330208627564597/1315306732657971240

**Comment by antlio** [comment]
@jpjpjp thanks for reporting this! tried to fix it recently but need more love i guess. 

do you happen to have a piece of specification to share here?

**Comment by jpjpjp** [comment]
You can see this issue in action here: https://lm-v2-api-v1-proxy-08a8f2bf0c0c.herokuapp.com/#tag/transactions/DELETE/transactions.   

The spec is here: https://lm-v2-api-mock-data-f24357049a1b.herokuapp.com/v2/openapi/

Let me know if you have any problems reproducing!

**Comment by antlio** [comment]
@jpjpjp i just tried and it seems to work now [in your reference](https://lm-v2-api-v1-proxy-08a8f2bf0c0c.herokuapp.com/#tag/transactions/DELETE/transactions) - can you confirm?

**Comment by antlio** [comment]
however we can keep this open as the dropdown from the reference is not reflecting the update from the modal, will look at it!

**Comment by jpjpjp** [comment]
@antlio yes exactly.  The functionality works, it's just that the name in the dropdown doesn't update when you pick the 2nd or 3rd example.

**Comment by antlio** [comment]
@jpjpjp can you try again? i don't reproduce on my end when picking 2-3rd examples - title changes well

**Comment by jpjpjp** [comment]
Hi @antlio, yes, seems to be working fine now for me as well.   I'm not sure what has changed.   I can close this issue if you'd like.
